tests/Database/BuilderDatabaseFunctionalTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace MongoDB\Tests\Database;
+
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+
+use function iterator_to_array;
+
+class BuilderDatabaseFunctionalTest extends FunctionalTestCase
+{
+    public function tearDown(): void
+    {
+        $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
+
+        parent::tearDown();
+    }
+
+    public function testAggregate(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::documents([
+                ['x' => 1],
+                ['x' => 2],
+                ['x' => 3],
+            ]),
+            Stage::bucketAuto(
+                groupBy: Expression::intFieldPath('x'),
+                buckets: 2,
+            ),
+        );
+        // Extract the list of stages for arg type restriction
+        $pipeline = iterator_to_array($pipeline);
+
+        $results = $this->database->aggregate($pipeline)->toArray();
+        $this->assertCount(2, $results);
+    }
```

Comment:
Failing test with `test-mongodb-5.0-replicaset-noauth-nossl`.
```
[2024/09/11 12:21:08.760] ERROR:  (MongoDB\Driver\Exception\CommandException)
[2024/09/11 12:21:08.760] MongoDB\Tests\Database\BuilderDatabaseFunctionalTest::testAggregate
[2024/09/11 12:21:08.760] MongoDB\Driver\Exception\CommandException: $documents' is not allowed in user requests
[2024/09/11 12:21:08.760] /data/mci/4db91d1c8fc44dbdea82e2266907468c/src/src/Operation/Aggregate.php:334
[2024/09/11 12:21:08.760] /data/mci/4db91d1c8fc44dbdea82e2266907468c/src/src/Operation/Aggregate.php:241
[2024/09/11 12:21:08.760] /data/mci/4db91d1c8fc44dbdea82e2266907468c/src/src/Database.php:245
[2024/09/11 12:21:08.760] /data/mci/4db91d1c8fc44dbdea82e2266907468c/src/tests/Database/BuilderDatabaseFunctionalTest.php:37
```

Reply:
`$documents` was added in MongoDB 6.0.